### PR TITLE
Prevent overwriting passes when data is passed in VW at object creation

### PR DESCRIPTION
**Pull request by Smit-create**
Partial fix for #1469
* Prevents overwriting  the `passes` when data is passed at object creation

### CODE REVIEW

**Comment by RituRajSingh878** [comment]
@Smit-create In this [comment ](https://github.com/VowpalWabbit/vowpal_wabbit/issues/1469#issuecomment-591191032)
I think @gramhagen is not allowing us to do this. Also, We should put` init` as clean as possible because I was [trying to pass](https://github.com/VowpalWabbit/vowpal_wabbit/issues/2333) the ` check_estimator`, and It is failing due to the extra variables created in the `__init__`. Also, I am pretty sure that the sklearn check_estimator will not allow doing this if we wish to pass the check_estimaot of sklearn So I think we should not add anything in `init.`

Let's see what others think?

**Comment by Smit-create** [comment]
So, going through the comment, I think that the case when the data file is passed as the parameters needs attention, to make it clearer that when to initialize a model, whether during the fit(here it is not required though), or directly during predict, as an option to this PR.

**Comment by Sharad24** [comment]
Comments over [here](https://github.com/VowpalWabbit/vowpal_wabbit/pull/2285#discussion_r383425664) might be a bit relevant :)

**Comment by Smit-create** [comment]
Sure, Thanks for the update on this.

**Comment by Smit-create** [comment]
I have updated the PR accordingly and removed model creation in __init__.

**Comment by Smit-create** [comment]
@jackgerrits Please review this.

File: python/vowpalwabbit/sklearn_vw.py [hunk 1]
```diff
@@ -344,7 +344,9 @@ def __init__(self,
         if any(x in self.params for x in ext_file_args):
             # fitting will be handled by vw directly
             self.fit_ = True
-            self.passes_ = 1
+            if not self.params.get('passes', None): # To prevent overwriting of passes
+                self.params['passes'] = 1   # default 1 if not given
+            self.passes_ = self.params['passes']
```
Comment (gramhagen) [inline python/vowpalwabbit/sklearn_vw.py]:
```suggestion
            self.passes_ = self.params.get('passes', 1)
```

Comment (Smit-create) [reply python/vowpalwabbit/sklearn_vw.py]: Oh, I am so sorry, it went completely out of my mind. Thanks :)

**Comment by Smit-create** [comment]
Does this look good to merge?

**Comment by Smit-create** [comment]
Thanks @gramhagen!

**Comment by Smit-create** [comment]
@jackgerrits Please merge if this good to go.

**Pull request merged**